app/v0/protocol_v0.go

```diff
@@ -114,16 +114,38 @@ func (p *ProtocolV0) GetVersion() uint64 {
 	return p.version
 }
 
-func (p *ProtocolV0) ValidateTx(ctx sdk.Context, txBytes []byte) sdk.Error {
+func (p *ProtocolV0) ValidateTx(ctx sdk.Context, txBytes []byte, msgs []sdk.Msg) sdk.Error {
+
+	serviceMsgMum := 0
+	for _, msg := range msgs {
+		if msg.Route() == service.MsgRoute {
+			serviceMsgMum++
+		}
+	}
+
+	if serviceMsgMum != 0 && serviceMsgMum != len(msgs) {
+		return sdk.ErrServiceTxLimit("Can't mix the service msg with other types of msg in a transaction! ")
+	}
+
 	subspace, bool := p.paramsKeeper.GetSubspace(auth.DefaultParamSpace)
 	var txSizeLimit uint32
 	if bool {
 		subspace.Get(ctx, auth.TxSizeLimitKey, &txSizeLimit)
 	} else {
 		panic("The subspace " + auth.DefaultParamSpace + " cannot be found!")
 	}
+	if serviceMsgMum == 0 && uint32(len(txBytes)) > txSizeLimit {
+		return sdk.ErrExceedsTxSize("the tx size exceeds the limitation")
+	}
 
-	if uint32(len(txBytes)) > txSizeLimit {
+	subspace, bool = p.paramsKeeper.GetSubspace(service.DefaultParamSpace)
+	var serviceTxSizeLimit uint64
```

Comment:
We need to get the service_txsize_limit param only if `serviceMsgMum == len(msgs)`